File: backend/src/internal/models/notification.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Notification struct {
	ID           uint            `json:"id" gorm:"primaryKey"`
	Title        string          `json:"title" gorm:"size:100;unique" validate:"required,min=2"`
	Description  string          `json:"description" gorm:"type:text"`
	OrderNum     uint            `json:"orderNum" gorm:"default:0"`
	Status       bool            `json:"status"`
	MainImage    string          `json:"mainImage" gorm:"type:text"`
	ImageUpload  string          `json:"imageUpload" gorm:"type:text"`
	NoticeType   string          `json:"noticeType"`
	Level        uint            `json:"level" gorm:"default:0"`
	DomainID     *uint           `json:"domainId" gorm:"index"`
	Domain       *Domain         `json:"domain" gorm:"constraint:OnUpdate:CASCADE,OnDelete:SET NULL;"`
	RegisterDate time.Time       `json:"registerDate"`
	Views        uint            `json:"views"`
	CreatedAt    time.Time       `json:"createdAt"`
	UpdatedAt    time.Time       `json:"updatedAt"`
	DeletedAt    *gorm.DeletedAt `gorm:"index" json:"deletedAt,omitempty"`
}

File: frontend/src/src/actions/notification.ts
import { gql } from "@apollo/client";

export const FILTER_NOTI = gql`
  query GetNotifications(
    $filters: [Filter!]
    $orders: [Order!]
    $pagination: Pagination
  ) {
    response: getNotifications(
      filters: $filters
      orders: $orders
      pagination: $pagination
    ) {
      notifications {
        id
        title
        description
        orderNum
        status
        level
        domainId
        domain {
          id
          name
        }
        mainImage
        imageUpload
        noticeType
        registerDate
        views
        createdAt
        updatedAt
        deletedAt
      }
      total
    }
  }
`;

export const CREATE_NOTI = gql`
  mutation NewNotification($input: NewNotificationInput!) {
    response: createNotification(input: $input) {
      id
      title
      description
      mainImage
      imageUpload
      noticeType
      registerDate
      views
    }
  }
`;

export const UPDATE_NOTI = gql`
  mutation UpdateNoti($id: ID!, $input: UpdateNotificationInput!) {
    response: updateNotification(id: $id, input: $input) {
      id
      title
      description
      mainImage
      imageUpload
      noticeType
      registerDate
      level
      orderNum
      views
      updatedAt
      createdAt
      deletedAt
    }
  }
`;

export const DELETE_NOTI = gql`
  mutation DeleteNotification($id: ID!) {
    response: deleteNotification(id: $id)
  }
`;
